s3torchconnector/src/s3torchconnector/s3iterable_dataset.py

```diff
@@ -26,30 +26,38 @@ def __init__(
         self,
         region: str,
         get_dataset_objects: Callable[[S3Client], Iterable[S3BucketKey]],
+        endpoint: str = "",
         transform: Callable[[S3Reader], Any] = identity,
     ):
         self._get_dataset_objects = get_dataset_objects
         self._transform = transform
         self._region = region
+        self._endpoint = endpoint
         self._client = None
 
     @property
     def region(self):
         return self._region
 
+    @property
+    def endpoint(self):
+        return self._endpoint
+
     @classmethod
     def from_objects(
         cls,
         object_uris: Union[str, Iterable[str]],
         *,
         region: str,
+        endpoint: str = "",
```

Comment:
Same here for default `None`

Reply:
Changed the default value of endpoint parameter to None instead of empty string